File: 2195-time-needed-to-buy-tickets/time-needed-to-buy-tickets.py
class Solution:
    #   Iterate and increase 'elapsed_time' if
    #   person still has tickets to purchase
    #   (tracked by 'cycle'). Return when person
    #   buys last ticket
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        elapsed_time = 0

        for cycle in range(1, tickets[k] + 1):
            for idx, val in enumerate(tickets):
                if val >= cycle:
                    elapsed_time += 1
                    if idx == k and val == cycle:
                        return elapsed_time

        return 0

File: 2195-time-needed-to-buy-tickets/time-needed-to-buy-tickets.c
int timeRequiredToBuy(int* tickets, int ticketsSize, int k) {
    //  Iterate and increase 'time' if person
    //  still has tickets to purchase (tracked by
    //  'cycle'). Return when person buys last
    //  ticket
    int time = 0;

    for (int i = 1; i <= tickets[k]; i++) {
        for (int j = 0; j < ticketsSize; j++) {
            if (tickets[j] >= i) {
                time++;
                if (j == k && i == tickets[k])
                    return time;
            }
        }
    }

    return 0;
}